pkg/backporter/backporter.go

```diff
@@ -159,214 +168,271 @@ const clonesTemplateConstructor = `
+			return
 		}
+
 		jsonBugInfo, err := json.MarshalIndent(*bugInfo, "", "  ")
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			_, fprintfErr := fmt.Fprintf(w, "failed to marshal bugInfo to JSON: %v", err)
-			if fprintfErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
-				return fprintfErr
-			}
-			// logger.WithError(err).Errorf("failed to marshal bugInfo to JSON")
-			return err
+			http.Error(w, "failed to marshal bugInfo to JSON", http.StatusInternalServerError)
+			logrus.WithFields(logFieldsFor(endpoint, bugID)).WithError(fmt.Errorf("failed to marshal bugInfo to JSON: %w", err))
+			return
 		}
+
 		w.WriteHeader(http.StatusOK)
 		_, err = w.Write(jsonBugInfo)
 		if err != nil {
-			http.Error(w, "Error while building page", http.StatusInternalServerError)
-			return err
+			http.Error(w, "unable to write to responsewriter for getBugHandler", http.StatusInternalServerError)
+			logrus.WithFields(logFieldsFor(endpoint, bugID)).WithError(fmt.Errorf("unable to write to responsewriter for getBugHandler: %w", err))
+			return
+		}
+	}
+}
+
+func getClonesTemplateData(bugID int, client bugzilla.Client, allTargetVersions sets.String) (*ClonesTemplateData, int, error) {
+	bug, err := client.GetBug(bugID)
+	if err != nil {
+		return nil, http.StatusNotFound, fmt.Errorf("Bug#%d not found: %w", bugID, err)
+	}
+
+	prs, err := client.GetExternalBugPRsOnBug(bug.ID)
+	if err != nil {
+		return nil, http.StatusInternalServerError, fmt.Errorf("Bug#%d - error occured while retreiving list of PRs: %w", bug.ID, err)
+	}
+
+	parent, err := client.GetRootForClone(bug)
+	if err != nil {
+		return nil, http.StatusInternalServerError, fmt.Errorf("unable to fetch root bug: %w", err)
+	}
+
+	clones, err := client.GetAllClones(bug)
+	if err != nil {
+		return nil, http.StatusInternalServerError, fmt.Errorf("unable to get clones: %w", err)
+	}
+	// Target versions would be used to populate the CreateClone dropdown
+	targetVersions := sets.NewString(allTargetVersions.List()...)
+	// Remove target versions of the original bug
+	targetVersions.Delete(bug.TargetRelease...)
+	for _, clone := range clones {
+		clonePRs, err := client.GetExternalBugPRsOnBug(clone.ID)
+		// Remove target releases which already have clones
+		targetVersions.Delete(clone.TargetRelease...)
+
+		if err != nil {
+			return nil, http.StatusInternalServerError, fmt.Errorf("Bug#%d - error occured while retreiving list of PRs : %w", clone.ID, err)
+		}
+		clone.PRs = clonePRs
+	}
+	wrpr := ClonesTemplateData{
+		Bug:          bug,
+		Clones:       clones,
+		Parent:       parent,
+		PRs:          prs,
+		CloneTargets: targetVersions.List(),
+	}
+	return &wrpr, http.StatusOK, nil
+}
+
+// ClonesHandler acts as a router for the RESTish calls to clones
+func ClonesHandler(client bugzilla.Client, allTargetVersions sets.String) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
```

Comment:
Can we move this check into `GetClonesHandler`? Why is it here, are we expecting to handle more verbs here in the future?

Reply:
I had added it for the POST clones which steve had mentioned and left it in there if we were going to move it back. I will remove it since we are sticking with /clones/create